File: scraper.py
import os
os.environ["SCRAPERWIKI_DATABASE_NAME"] = "sqlite:///data.sqlite"

import scraperwiki
import requests
from datetime import datetime
from bs4 import BeautifulSoup

today = datetime.today()

url = 'https://www.geelongaustralia.com.au/advertisedplanning/default.aspx'
r = requests.get(url)
soup = BeautifulSoup(r.content, 'lxml')

rows = soup.find_all('tr')

for row in rows:
    try:
        record = {}
        suburb = row.find_all('td')[1].text
        record['address'] = f'{row.find_all("td")[0].text}, {suburb}'.strip().title()
        record['date_scraped'] = today.strftime("%Y-%m-%d")
        info_url = 'https://www.geelongaustralia.com.au/advertisedplanning/' + \
            str(row.find_all('td')[0]).split('"')[1]
        rr = requests.get(info_url)
        soupp = BeautifulSoup(rr.content, 'lxml')
        record['description'] = soupp.find_all('p')[3].text.strip()
        record['council_reference'] = row.find_all('td')[2].text
        record['info_url'] = info_url
        on_notice_from_raw = row.find_all('td')[3].text
        record['on_notice_from'] = datetime.strptime(
            on_notice_from_raw, '%d %b %Y').strftime("%Y-%m-%d")
        on_notice_to_raw = row.find_all('td')[4].text
        record['on_notice_to'] = datetime.strptime(
            on_notice_to_raw, '%d %b %Y').strftime("%Y-%m-%d")

        scraperwiki.sqlite.save(
            unique_keys=['council_reference'], data=record, table_name="data")
    except:
        continue

File: scraper.rb
require 'scraperwiki'
require 'mechanize'

FileUtils.touch('data.sqlite')

today = Time.now.strftime('%Y-%m-%d')

url   = 'https://www.geelongaustralia.com.au/advertisedplanning/default.aspx'
agent = Mechanize.new
page  = agent.get(url)

table1 = page.search('table.table.table-striped')[0]
table1.search('tr')[0].remove
rows1 = table1.search('tr')
table2 = page.search('table.table.table-striped')[1]
table2.search('tr')[0].remove
rows2 = table2.search('tr')

for row in rows1 do
  record = {}
  suburb = row.search('td')[1].text.strip
  record['address'] = row.search('td')[0].text.strip + ', ' + suburb
  record['council_reference'] = row.search('td')[2].text.strip
  record['date_scraped'] = today
  record['on_notice_from'] = DateTime.strptime(row.search('td')[3].text.strip, '%d %b %Y').strftime('%Y-%m-%d')
  record['on_notice_to'] = DateTime.strptime(row.search('td')[4].text.strip, '%d %b %Y').strftime('%Y-%m-%d')
  link = 'https://www.geelongaustralia.com.au/advertisedplanning/' + row.search('a').to_s.split('"')[1]
  record['info_url'] = link
  page  = agent.get(link)
  record['description'] = page.search('p')[3].text.strip
  ScraperWiki.save_sqlite(['council_reference'], record)
end

for row in rows2 do
  record = {}
  suburb = row.search('td')[1].text.strip
  record['address'] = row.search('td')[0].text.strip + ', ' + suburb
  record['council_reference'] = row.search('td')[2].text.strip
  record['date_scraped'] = today
  record['on_notice_to'] = DateTime.strptime(row.search('td')[3].text.strip, '%d %b %Y').strftime('%Y-%m-%d')
  link = 'https://www.geelongaustralia.com.au/advertisedplanning/' + row.search('a').to_s.split('"')[1]
  record['info_url'] = link
  page  = agent.get(link)
  record['description'] = page.search('p')[3].text.strip
  ScraperWiki.save_sqlite(['council_reference'], record)
end
